src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt

```diff
@@ -341,21 +346,24 @@ class JooqBurdenEstimateRepository(
         val (set, _) = getSetFromResponsibilityPath(groupId, touchstoneId, scenarioId, setId)
         if (set.status == BurdenEstimateSetStatus.COMPLETE)
         {
-            throw OperationNotAllowedError("You cannot clear a burden estimate set which is marked as 'complete'.")
+            throw InvalidOperationError("You cannot clear a burden estimate set which is marked as 'complete'.")
         }
 
         // We do this first, as this change will be rolled back if the annex
         // changes fail, but the reverse is not true
         changeBurdenEstimateStatus(setId, BurdenEstimateSetStatus.EMPTY)
+        getEstimateWriter(set).clearEstimateSet(setId)
+    }
 
-        val type = set.type.type
-        if (type == BurdenEstimateSetTypeCode.STOCHASTIC)
+    private fun getEstimateWriter(set: BurdenEstimateSet): BurdenEstimateWriter
+    {
+        return if (set.type.type == BurdenEstimateSetTypeCode.STOCHASTIC)
```

Comment:
could use the `.isStochastic()` helper

Reply:
Not quite, because it exists on `CreateBurdenEstimateSet`, not `BurdenEstimateSet`. However, I've now added a similar method: https://github.com/vimc/montagu-api/pull/233/commits/fe9971478f1a585caab62542489d2b38a0f927a7

You'll need to also merge this small webmodels PR https://github.com/vimc/montagu-webmodels/pull/41 and then return the API to webmodels master.

Reply:
Ah I meant `type.isStochastic` but extra helper method is even better :)